### PR TITLE
Added option to Appointment  Display Sets

### PR DESCRIPTION
**Pull request by epsdky**
Option added to Appointment Display Sets to allow the last set to expand past the appointment display limit.

### CODE REVIEW

File: interface/super/edit_globals.php [hunk 1]
```diff
@@ -306,6 +306,7 @@ function checkBackgroundServices(){
 <script type="text/javascript" src="../../library/js/jquery.1.3.2.js"></script>
 <script type="text/javascript" src="../../library/js/common.js"></script>
 <script type="text/javascript" src="../../library/js/fancybox/jquery.fancybox-1.2.6.js"></script>
+<script type="text/javascript" src="<?php echo $GLOBALS['assets_static_relative']; ?>/jscolor-1-4-5/jscolor.js"></script>
```
Comment (bradymiller) [inline interface/super/edit_globals.php]: whoohooo,
nice to see this being used

File: interface/patient_file/summary/demographics.php [hunk 2]
```diff
@@ -1347,28 +1347,77 @@ function setMyPatient() {
...
+                  if($events[$limitApptIndx - $i]['pc_eventDate'] != $limitApptDate) {
+                    $firstApptIndx = $apptNum2 - $i;
+                    break;
+                  }
+                }
+                //
+          }
+          //
+          if($extraApptDate) {
+            if($extraApptDate != $limitApptDate) $apptStyle2 = " style='background-color:$colorSet3;'";
```
Comment (bradymiller) [inline interface/patient_file/summary/demographics.php]: attr() the $colorSet3

File: interface/patient_file/summary/demographics.php [hunk 3]
```diff
@@ -1347,28 +1347,77 @@ function setMyPatient() {
...
+            else $apptStyle2 = " style='background-color:$colorSet4;'";
```
Comment (bradymiller) [inline interface/patient_file/summary/demographics.php]: attr() the $colorSet4

File: interface/patient_file/summary/demographics.php [hunk 4]
```diff
@@ -1400,15 +1449,14 @@ function setMyPatient() {
...
-              else $apptStyle = " style='background-color:#e6ffe6;'";
+              $apptStyle = $apptStyle2;
             } else {
               if($row['pc_eventDate'] != $priorDate) {
                 $priorDate = $row['pc_eventDate'];
                 $toggleSet = !$toggleSet;
               }
-              if($toggleSet) $apptStyle = " style='background-color:#e6e6ff;'";
-              else $apptStyle = '';
+              if($toggleSet) $apptStyle = " style='background-color:$colorSet2;'";
```
Comment (bradymiller) [inline interface/patient_file/summary/demographics.php]: attr()

File: interface/patient_file/summary/demographics.php [hunk 5]
```diff
@@ -1400,15 +1449,14 @@ function setMyPatient() {
...
+              else $apptStyle = " style='background-color:$colorSet1;'";
```
Comment (bradymiller) [inline interface/patient_file/summary/demographics.php]: attr()

File: interface/patient_file/summary/demographics.php [hunk 6]
```diff
@@ -1427,7 +1475,7 @@ function setMyPatient() {
             if ( $count < 1 ) {
                 echo "&nbsp;&nbsp;" . htmlspecialchars(xl('None'),ENT_NOQUOTES);
             } else { //////
-              if($extraApptDate) echo "<div>&nbsp;</div>";
+              if($extraApptDate) echo "<div style='color:#0000cc;'><b>$extraApptDate ( + ) </b></div>";
```
Comment (bradymiller) [inline interface/patient_file/summary/demographics.php]: text() around the $extraApptDate

File: interface/super/edit_globals.php [hunk 7]
```diff
@@ -532,6 +533,20 @@ function validate_file(){
       echo "  </select>\n";
     }
 
+    else if ($fldtype == 'color_code') {
+      if ($_GET['mode'] == "user") {
+        $globalTitle = $globalValue;
+      }
+      echo "  <input type='text' class='color {hash:true}' name='form_$i' id='form_$i' " .
+        "size='6' maxlength='15' value='$fldvalue' />" .
```
Comment (bradymiller) [inline interface/super/edit_globals.php]: attr()

File: interface/super/edit_globals.php [hunk 8]
```diff
@@ -532,6 +533,20 @@ function validate_file(){
...
+        "<input type='button' value='Default' onclick='resetDefault_$i()'>";
+      echo "\n<script>\n" .
+        "function resetDefault_$i() {\n" .
+        "document.forms[0].form_$i.color.fromString('$flddef');\n" .
```
Comment (bradymiller) [inline interface/super/edit_globals.php]: attr()

File: library/globals.inc.php [hunk 9]
```diff
@@ -1232,6 +1232,41 @@
       xl('Number of Appointments to display in the Onsite Patient Portal')
     ),
 
+    'appt_display_sets_option' => array(
+      xl('Appointment Display Sets - Ignore Display Limit (Last Set)'),
+      'bool',                           // data type
+      '1',                              // default
+      xl('Appointment Display Sets - Allow Last Set to Expand Past Appointment Display Limit')
```
Comment (bradymiller) [inline library/globals.inc.php]: Do you mean will always shows all appt on same day even if over limit. Or mean something else?

File: library/globals.inc.php [hunk 10]
```diff
@@ -1232,6 +1232,41 @@
...
+      '#E6FFE6',
+      xl('Appointment Display Sets - Color - Set Three')
+    ),
+
+    'appt_display_sets_color_4' => array(
+      xl('Appointment Display Sets - Color - Set Four'),
+      'color_code',
+      '#FFE6FF',
+      xl('Appointment Display Sets - Color - Set Four')
+    ),
```
Comment (bradymiller) [inline library/globals.inc.php]: Just to have an idea, what do these 4 different sets represent?

**Comment by bradymiller** [comment]
Placed review. Will test it out on the next revision.

File: library/globals.inc.php (hunk 9 above)
Comment (epsdky) [inline library/globals.inc.php]: Yes.

File: library/globals.inc.php (hunk 10 above)
Comment (epsdky) [inline library/globals.inc.php]: I have rewritten the descriptions and will add them...

Color for odd sets (except when the last set is odd and the number of appointments scheduled exceeds the appointment display limit)

Color for even sets (except when the last set displayed is even and the number of appointments scheduled exceeds the appointment display limit)

Color for the last set displayed when the number of appointments scheduled exceeds the display limit and all appointments for the associated date are shown

Color for the last set displayed when the number of appointments scheduled exceeds the display limit and not all appointments for the associated date are shown

File: interface/super/edit_globals.php (hunk 1 above)
Comment (epsdky) [inline interface/super/edit_globals.php]: :smile:

**Comment by epsdky** [comment]
Hello Brady,

I have written a description in case you want to pull in the above code


Extended Appointment Display Sets (Patient Summary)


Option added to Appointment Display Sets to allow (if necessary) the appointment display limit to be disregarded so that all appointments can be displayed for the last set.

for Appointment Display Sets appointments with same date are grouped into sets for display.

The last set turns green (default) to indicate that all member appointments are diplayed and at least one subsequent scheduled appointment exists (not displayed).

The last set turns pink (default) when not all member appointments are displayed.

**Comment by bradymiller** [comment]
Just committed this to the codebase. Thanks for the contribution!

**Pull request closed without merging**